subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/internal/EclipseProjectMetadata.java

```diff
@@ -25,18 +27,18 @@
 import java.util.Set;
 
 public class EclipseProjectMetadata implements IdeProjectMetadata {
-    private final EclipseProject eclipseProject;
+    private final Provider<EclipseProject> eclipseProject;
     private final File projectDir;
-    private final Task generatorTask;
+    private final TaskProvider<? extends Task> generatorTask;
 
-    public EclipseProjectMetadata(EclipseProject eclipseProject, File projectDir, Task generatorTask) {
+    public EclipseProjectMetadata(Provider<EclipseProject> eclipseProject, File projectDir, TaskProvider<? extends Task> generatorTask) {
```

Comment:
The project should be owned by the model and be eagerly created, and so we should get rid of the provider here.